lib/core/versions/latest/AnchorFile.ts

```diff
@@ -61,6 +63,13 @@ export default class AnchorFile {
       throw new SidetreeError(ErrorCode.AnchorFileMissingOperationsProperty);
     }
 
+    // This property is optional so if there's an additional propertye besides
+    // the required ones then it must be this optional one.
+    if (anchorFileProperties.length === maxAllowedPropertyCount &&
+       !anchorFileModel.hasOwnProperty('writerLock')) {
+      throw new SidetreeError(ErrorCode.AnchorFileHasUnknownProperty);
```

Comment:
Nit: In principal we try to never reuse an error code, so consider either:

1. Create a new error code. or
1. Attach custom error message, or
1. Follow the pattern found in other validation handling optional parameters, e.g.
```javascript
    const allowedProperties = new Set(['batchFileHash', 'updateOperations']);
    for (let property in mapFile) {
      if (!allowedProperties.has(property)) {
        throw new SidetreeError(ErrorCode.MapFileHasUnknownProperty);
      }
    }
```

Reply:
Really good point. Will update.